virtual/registry/sql_kv.go

```diff
@@ -0,0 +1,194 @@
+package registry
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+)
+
+// TODO: hardcoded for now, but may be useful as a configuration option for
+// some type of isolation
+const defaultTableName = "nola_kv"
+
+type sqlKV struct {
+	db *sql.DB
+	sqlKVStatement
+}
+
+type sqlKVStatement struct {
+	tableName  string
+	vst        time.Time
+	setStmt    *sql.Stmt
+	getStmt    *sql.Stmt
+	deleteStmt *sql.Stmt
+	rangeStmt  *sql.Stmt
+}
+
+func newSQLKV(driver, dsn string) (kv, error) {
+	tableName := defaultTableName
+	db, err := sql.Open(driver, dsn) // TODO support selecting driver at runtime
+	if err != nil {
+		return nil, err
+	}
+
+	db.SetMaxOpenConns(1)
+	db.SetMaxIdleConns(1)
+	db.SetConnMaxIdleTime(0)
+
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO Improve indexing
```

Comment:
What can be improved here? Just curious if you have any ideas

Reply:
I added a comment in here suggesting that we could add an index on the first few bytes of the index to improve the performance of iterPrefix.